src/python/WMCore/MicroService/MSRuleCleaner/MSRuleCleaner.py

```diff
@@ -317,9 +317,21 @@ def _dispatchWflow(self, wflow):
             #       {REPLICATING, STUCK, SUSPENDED, WAITING_APPROVAL}.)
             #       We still need some proper logging for them.
             msg = "Skipping workflow: %s - tape transfers are not yet completed." % wflow['RequestName']
-            msg += " Will retry again in the next cycle."
+            msg += " Will retry again in the next cycle. deleting the MS-Transferor rules"
             self.logger.info(msg)
             self._checkStatusAdvanceExpired(wflow, additionalInfo=msg)
+            for pline in self.mstrlines:
+                try: 
+                    pline.run(wflow)
+                except Exception as ex:
+                    msg = "%s: General error from pipeline while cleaning Transfer rukes. Workflow: %s. Error:  \n%s. " % (pline.name, wflow['RequestName'], str(ex))
+                    msg += "\nWill retry again in the next cycle."
+                    self.logger.exception(msg)
+                    self._checkStatusAdvanceExpired(wflow, additionalInfo=msg)
```

Comment:
The call to `self._checkStatusAdvanceExpired` is redundant here. It already happens on line 322. You may safely remove this one here.